lib/pbench/server/api/resources/users_api.py

```diff
@@ -469,20 +470,23 @@ def delete(self, username):
             abort(403, message="Not authorized to delete user")
 
         try:
-            user = User.query(username=username)
-            # Do not delete if the user is admin
-            if not user.is_admin():
-                User.delete(username)
+            User.delete(username)
+            self.logger.info("User entry deleted for user with username {}", username)
+        except NoResultFound:
+            self.logger.info(
+                "Delete operation on non-existent user '{}'", username,
+            )
+            abort(404, message="User not found")
+        except AdminUserDeletion:
+            self.logger.info(
+                "Delete operation on an admin user '{}'", username,
```

Comment:
Given you have an explicit `__str__()` method for the `AdminUserDeletion`, could we use that with something like:

```
except AdminUserDeletion as exc:
    self.logger.info(str(exc))
    abort(403, message=f"...")
```